src/views/LoginSuccess.vue

```diff
@@ -30,226 +29,108 @@ import { Component, Emit, Vue } from 'vue-property-decorator';
+import Loader from '@/components/Loader.vue';
+import WalletInfoCollector from '@/lib/WalletInfoCollector';
 
-@Component({components: {PageHeader, PageBody, LabelInput, AccountList, Network, PageFooter, SmallPage}})
+@Component({components: {PageHeader, PageBody, LabelInput, AccountList, Loader, PageFooter, SmallPage}})
 export default class LoginSuccess extends Vue {
     @Static private request!: ParsedLoginRequest;
     @State private keyguardResult!: ImportResult;
 
-    private keyguard!: KeyguardClient;
-    private network!: Network;
-
     private walletInfo: WalletInfo | null = null;
 
-    private walletLabel: string = 'Keyguard Wallet';
-    private defaultAccountLabel: string = 'Standard Account';
-
     /**
-     * Maps are not supported by Vue's reactivity system, thus updates to a Map do not trigger
-     * an update of the DOM. To work around that, a second reactive attribute is added (a plain
-     * number) which is incremented each time the Map is updated. The DOM is then fed not by the
-     * Map itself, but by a computed property (`this.accountsArray` in this case), which uses
-     * both the non-reactive Map and the reactive partner-attribute. The partner-attribute change
-     * triggers the recomputation, which in turn updates the DOM.
+     * Used to invalidate the cached value for getter `accountsArray`.
      */
-    private accounts: Map<string, AccountInfo> = new Map();
-    private accountsUpdateCount: number = 0;
+    private updateCount: number = 0;
 
-    private lastDerivedIndex: number = 0;
-    private lastActiveIndex: number = 0;
-
-    private result?: LoginResult;
-
-    private contentReady: boolean = false;
+    private retrievalFailed: boolean = false;
     private retrievalComplete: boolean = false;
 
     private async mounted() {
-        const currentWalletInfo = await WalletStore.Instance.get(this.keyguardResult.keyId);
-
-        // if there is a WalletInfo for the returned keyId already, use it
-        if (currentWalletInfo && currentWalletInfo.label) {
-            this.walletLabel = currentWalletInfo.label;
-            if (currentWalletInfo.accounts) {
-                this.accounts = currentWalletInfo.accounts;
-            }
-        }
-
         // The Keyguard always returns (at least) one derived Address,
-        // thus we can already create a complete WalletInfo object that
-        // can be displayed while waiting for the network.
-        // Before adding the WalletInfo object, make sure it is not set already.
-        this.keyguardResult.addresses.forEach((addressObj) => {
-            const address = new Nimiq.Address(addressObj.address);
-            if (!this.accounts.has(address.toUserFriendlyAddress())) {
-                const addressInfo = new AccountInfo(
-                    addressObj.keyPath,
-                    this.defaultAccountLabel,
-                    address,
+        const keyguardResultAccounts = this.keyguardResult.addresses.map((addressObj) => ({
```

Comment:
I'm guessing this should be a `BasicAccountInfo[]` type? Could you add the annotation for it?

Reply:
It is a `BasicAccountInfo[]` yes, but that type is currently only available internally in `WalletInfoCollector` and I'm not much in favor of exporting it as exporting some sort of `AccountInfo` from a `WalletInfoCollector` sounds weird to me. It would rather be something like `WalletInfoCollectorIntermediateAccount` which is not a snappy name. Also I don't like having several account info types that are about the same.
I think keeping it the way it is best. If the types don't match, TypeScript is already gonna recognize this.